src/simulation.jl

```diff
@@ -165,12 +282,47 @@ end
     add_responses!(signal, responses::Vector, e, s, tvec, erpvec)
     add_responses!(signal, responses::Matrix, e, s, tvec, erpvec)
     add_responses!(signal, responses::AbstractArray, e, s, tvec, erpvec)
-Helper function to add inplace the responses to the signal, but for both 2D (1 channel) and 3D (X channel case).
+
+Add (in-place) the given `responses` to the `signal`, for both 2D (1 channel) and 3D (X channel case). Helper function.
+
+# Arguments
+- `signal`: Continuous EEG signal to be modified in place. Has the dimensions `channels x continuous_time x subjects`.
+- `responses::Vector/Matrix/AbstractArray`: Responses to be added, has the dimensions `maxlength(components) x length(simulation.design)`. The data for all the subjects and their respective trials is concatenated.
```

Comment:
```suggestion
- `responses::Union{Vector, Matrix, AbstractArray}`: Responses to be added. In the multi-channel case, the dimensions are `channels x maxlength(components) x length(simulation.design)`, else `maxlength(components) x length(simulation.design)`.  The data for all the subjects and their respective trials is concatenated.
```
- I also encountered a similar case in one of the other docstrings and decided to use a type union (https://docs.julialang.org/en/v1/manual/types/#Type-Unions).
- I added the dimensions for the case when there are multiple channels.